src/classes/BDI_DataImportDeleteBTN_CTRL.cls

```diff
@@ -69,19 +96,39 @@ public with sharing class BDI_DataImportDeleteBTN_CTRL {
     }
 
     /** 
-     * @description Delete Data Import records
-     * @param isDeleteAll If true, delete all Data Import records, otherwise delete imported records only
+     * @description Finds Data Import records to be deleted
+     * @param isDeleteAll If true, find all Data Import records, otherwise find imported records only
      * @return PageReference Page specified in the 'retURL' parameter or Home page
      */
-    PageReference deleteDIRecords(Boolean isDeleteAll) {
-        try {
-            if (isDeleteAll) {
-                delete [SELECT Id FROM DataImport__c LIMIT 10000];
-            } else {
-                delete [SELECT Id FROM DataImport__c WHERE Status__c = :BDI_DataImport_API.bdiImported LIMIT 10000];
-            }
+    private PageReference findDIRecords(Boolean isDeleteAll) {
+        if (isDeleteAll) {
+            dataImports = [SELECT Id FROM DataImport__c LIMIT 10000];
```

Comment:
Rather than hardcode 10000, what about calculating this dynamically. Though theoretically the value should be 10000, this allows the action to evolve if Salesforce ever increases that limit.
`Integer maxRecords = Limits.getLimitDMLRows() - Limits.getDMLRows();`

Reply:
@force2b - good idea, fixed!